xcube/core/store/typespecifier.py

```diff
@@ -30,13 +30,14 @@
 import geopandas as gpd
 import xarray as xr
 
-class TypeId:
+class TypeSpecifier:
     """
-    A type id denotes a type of data. It is used to group similar types of data and discern different types of data.
-    It can be used by stores to state what types of data can be read from and/or written to them.
+    A type specifier denotes a type of data. It is used to group similar types of data and discern
```

Comment:
```suggestion
    A type specifier denotes a type of data. It is used to group similar types of data and distinguish between
```

I think this is clearer (hopefully I interpreted the intended meaning of "discern" correctly enough for a successful rephrasing...).